### PR TITLE
Add unittest and documentations for BDK configuration

### PR DESCRIPTION
**Pull request by symphony-hong**
Add unittest and documentations for BDK configuration

Move BdkConfigException to com.symphony.bdk.core.config.exceptions
to avoid many types of exceptions in a single packages

Add a condition in BdkConfigParser#parse to check if invalid YAML config is parsed to a single
text node.

### CODE REVIEW

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigParser.java [hunk 1]
```diff
@@ -35,7 +35,9 @@ public static JsonNode parse(InputStream inputStream) throws BdkConfigException
         }
 
         try {
-            return YAML_MAPPER.readTree(content);
+            JsonNode jsonNode =  YAML_MAPPER.readTree(content);
```
Comment (symphony-youri) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigParser.java:38]: We should probably make BdkConfigParser internal right? 
I see it is already does not have public visibility, maybe annotate it with API guardian?

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java [hunk 2]
```diff
@@ -1,53 +1,79 @@
...
+        InputStream inputStream = BdkConfigLoaderTest.class.getResourceAsStream("/config/config.json");
         BdkConfig config = BdkConfigLoader.loadFromInputStream(inputStream);
         assertEquals(config.getBot().getUsername(), "tibot");
     }
 
     @Test
-    public void loadFromYamlFileTest() throws JsonProcessingException, BdkConfigException {
-        String configPath = System.getProperty("user.dir") + "/src/test/resources/config.yaml";
+    public void loadFromYamlFileTest() throws BdkConfigException {
+        String configPath = System.getProperty("user.dir") + "/src/test/resources/config/config.yaml";
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:30]: I'm not sure that this is the safest way to locate the test resources. I'd personally use another strategy :
- using https://www.baeldung.com/junit-5-temporary-directory#1-tempdir-as-a-method-parameter
- copy the config file from the classpath to tempdir
- read config from tempdir

WDYT?

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java [hunk 3]
```diff
@@ -1,53 +1,79 @@
...
+        assertEquals(exception.getMessage(), "Config file is not found");
+    }
+
+    @Test
+    public void loadLegacyFromInputStreamTest() throws BdkConfigException {
+        InputStream inputStream = BdkConfigLoaderTest.class.getResourceAsStream("/config/legacy_config.json");
+        BdkConfig config = BdkConfigLoader.loadFromInputStream(inputStream);
+        assertEquals(config.getBot().getUsername(), "tibot");
+        assertEquals(config.getBot().getPrivateKeyPath(), "/Users/thibault.pensec/local/conf/agent/privatekey.pem");
+        assertEquals(config.getApp().getPrivateKeyPath(), "/Users/thibault.pensec/local/conf/agent/privatekey.pem");
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:69]: Please remove "thibault.pensec" from the test data 😅  I'll do the same with my PR !

File: symphony-bdk-core/src/test/resources/config/legacy_config.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "podHost": "devx1.symphony.com",
+  "agentHost": "devx1.symphony.com",
+  "keyAuthHost": "devx1.symphony.com",
+  "botUsername": "tibot",
+  "botPrivateKeyPath": "/Users/thibault.pensec/local/conf/agent/",
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/test/resources/config/legacy_config.json:6]: Please remove "thibault.pensec" !

File: symphony-bdk-core/src/test/resources/config/legacy_config.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "botPrivateKeyName": "privatekey.pem",
+  "appId": "tibapp",
+  "appPrivateKeyPath": "/Users/thibault.pensec/local/conf/agent/",
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/test/resources/config/legacy_config.json:9]: Same...

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java (hunk 2 above)
Comment (symphony-hong) [reply symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:30]: It's a good idea! I will try that way. 👍

**Pull request merged**